server/src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -105,6 +108,7 @@
  * Security-sensitive data such as passwords or private keys should not be stored in the cluster state, since the contents of the cluster
  * state are exposed in various APIs.
  */
+@SuppressForbidden(reason = "directly reading ClusterState#clusterFeatures")
```

Comment:
Let's not `SuppressForbidden` the whole class - instead introduce an accessor method that just calls the forbidden API, annotate it with `SuppressForbidden`, and include a comment about why this is ok.

Reply:
Yeah, this is a placeholder while I get the tests passing. It's all going inside `ClusterFeatures` itself in a min.